data_steward/tools/create_update_drc_id_match_table.py

```diff
@@ -34,12 +50,12 @@
 POPULATE_VALIDATION_TABLE = JINJA_ENV.from_string("""
 INSERT INTO `{{project_id}}.{{drc_dataset_id}}.{{id_match_table_id}}` (_PARTITIONTIME, {{fields}}) 
 SELECT TIMESTAMP_TRUNC(CURRENT_TIMESTAMP, HOUR), person_id, 
```

Comment:
I think this might need to be changed to something like
```suggestion
SELECT 
    CASE WHEN ABS(TIMESTAMP_TRUNC(CURRENT_TIMESTAMP, HOUR) - _PARTITIONTIME) < 2 THEN _PARTITIONTIME 
    ELSE TIMESTAMP_TRUNC(CURRENT_TIMESTAMP, HOUR) AS _PARTITIONTIME,
    person_id, 
```
My main concern is that the partition for PS will be created with a certain hour partition. But loading it with PS API data will take time due to API call limits. The id_match partition will only be created after the ps partition is fully populated. Is that correct? If so, the partition hour may be different, which can lead to problems while joining on the partition time. I think there may be more elegant ways to keep them in sync, so open to ideas. Let me know your thoughts.

Reply:
Interesting, that could be a problem. Yes, id_match is only created after ps_api load is done. We might be able to set the intended partition at the beginning of the python process (generate_metrics?) where they are both called, and then pass that partition explicitly into the SQL.